ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,248 @@
+import os
+import sys
+
+import aiokafka
+
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.constants import SPAN_KIND
+from ddtrace.constants import SPAN_MEASURED_KEY
+from ddtrace.contrib import trace_utils
+from ddtrace.ext import SpanKind
+from ddtrace.ext import SpanTypes
+from ddtrace.ext import kafka as kafkax
+from ddtrace.internal import core
+from ddtrace.internal.compat import time_ns
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.constants import MESSAGING_SYSTEM
+from ddtrace.internal.schema import schematize_messaging_operation
+from ddtrace.internal.schema import schematize_service_name
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils import set_argument_value
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.pin import Pin
+from ddtrace.propagation.http import HTTPPropagator as Propagator
+
+
+_AIOKafkaProducer = aiokafka.AIOKafkaProducer
+_AIOKafkaConsumer = aiokafka.AIOKafkaConsumer
+
+config._add(
+    "kafka",
+    dict(
+        _default_service=schematize_service_name("kafka"),
+        distributed_tracing_enabled=asbool(os.getenv("DD_KAFKA_PROPAGATION_ENABLED", default=False)),
+    ),
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(aiokafka, "__version__", "")
+
+
+class TracedAIOKafkaProducerMixin:
+    def __init__(self, *args, **kwargs):
+        super(TracedAIOKafkaProducerMixin, self).__init__(*args, **kwargs)
+        self._dd_bootstrap_servers = kwargs.get("bootstrap_servers")
+
+
+class TracedAIOKafkaConsumerMixin:
```

Comment:
This pattern looks like it was copied from the existing Kafka integration. It may work, but it's more complex than necessary. The confluent-kafka integration is written this way to work around the fact that native code can't be monkeypatched. Because aiokafka is pure Python, it can be monkeypatched with wrapt. I'd recommend that simpler approach here.

Reply:
Fixed